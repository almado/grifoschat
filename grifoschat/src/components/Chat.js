import React, { Component } from 'react';
import { List, ListItem, ListItemText, createGenerateClassName } from '@material-ui/core';
import _ from 'lodash';
import '../App.css';

class Chat extends Component {

    getClass(message) {
        console.log(message);
        return message.sender === "JoÃ£o" ? "ballon-primary" : "ballon-secondary";
    }

    render() {

        let msgs = [];

        this.props.messages.from.messages.map(msg => msgs.push(msg));
        this.props.messages.to.messages.map(msg => msgs.push(msg));

        msgs = _.sortBy(msgs, 'timestamp');
        console.log(this.props, 'merged', msgs);
        return (
            <List>
                {msgs.map(message => (
                    <li key={`section-${message.timestamp}`}>
                        <ul>
                            <div className={this.getClass(message)}>
                                {message.text}
                            </div>
                            {/* <ListItem key={`item-${message.timestamp}`}>
                                <ListItemText primary={message.text} className={this.getClass(message)} />
                            </ListItem> */}
                        </ul>
                    </li>
                ))}
            </List>
        );
    }
}

export default Chat;