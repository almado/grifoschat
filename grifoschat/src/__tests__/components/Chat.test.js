import React from 'react';
import { shallow, mount } from 'enzyme';

import Chat from '../../components/Chat';
import { wrap } from 'module';

describe('Message', () => {
    let wrapper;
    const INITIAL_STATE = {
        from: {
            avatar: './assets/maria-avatar.jpg',
            messages: [
                {
                    sender: 'Maria',
                    text: 'E ae... como vais!?',
                    timestamp: 1531863103149
                }
            ]
        },
        to: {
            messages: [
                {
                    sender: 'João',
                    text: 'E aewww maria! Foi mal a demora...',
                    timestamp: 1531864333332
                },
                {
                    text: 'Fala o que manda!',
                    timestamp: 1531864381645
                }
            ]
        }
    };

    beforeEach(() => {
        wrapper = shallow(<Chat messages={INITIAL_STATE} />);
    })

    it('Rendering match snapshot', () => {
        expect(Chat).toMatchSnapshot();
    });

    describe('Rendering props correctly', () => {
        let component;
        beforeEach(() => {
            component = mount(<Chat messages={INITIAL_STATE} />);
        });

        it('Props must be defined', () => {
            expect(component.props().messages).toBeDefined();
        });

        it('Rendering sender with value', () => {
            expect(component.props().messages.from.messages[0].sender).toBe('Maria');
            expect(component.props().messages.to.messages[0].sender).toBe('João');
        });

        it('Text should not be null or empty', () => {
            expect(component.props().messages.from.messages[0].text.length).toBeGreaterThanOrEqual(1);
            expect(component.props().messages.to.messages[0].text.length).toBeGreaterThanOrEqual(1);
        });
    });
});