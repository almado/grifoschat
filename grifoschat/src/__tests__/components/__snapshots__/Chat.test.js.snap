// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Rendering match snapshot 1`] = `[Function]`;
